controllers/openstackdataplanenodeset_controller.go

```diff
@@ -316,28 +328,117 @@ func (r *OpenStackDataPlaneNodeSetReconciler) Reconcile(ctx context.Context, req
 	return ctrl.Result{}, nil
 }
 
-func checkDeployment(helper *helper.Helper,
-	request ctrl.Request) (bool, bool, error) {
+func checkDeployment(
+	ctx context.Context,
+	helper *helper.Helper,
+	instance *dataplanev1.OpenStackDataPlaneNodeSet,
+	configMapHashes map[string]string,
+	secretHashes map[string]string,
+) (bool, bool, error) {
+
 	// Get all completed deployments
 	deployments := &dataplanev1.OpenStackDataPlaneDeploymentList{}
 	opts := []client.ListOption{
-		client.InNamespace(request.NamespacedName.Namespace),
+		client.InNamespace(instance.Namespace),
 	}
-	err := helper.GetClient().List(context.Background(), deployments, opts...)
+	err := helper.GetClient().List(ctx, deployments, opts...)
 	if err != nil {
 		helper.GetLogger().Error(err, "Unable to retrieve OpenStackDataPlaneDeployment CRs %v")
 		return false, false, err
 	}
+
+	hashNodeSetServices := false
+	deploymentExists := false
+	deploymentReady := false
+
 	for _, deployment := range deployments.Items {
```

Comment:
If we have multiple Deployment CRs for the same NodeSet does the hash calculation below considers that newest Deployment?

Reply:
All the hashes from all the Deployments are added. We're relying on the Watch to trigger this code. In theory the watch events ought to be ordered, as the Deployments are changed, so the latest would have been the most recent event. However, I assume there are numerous reasons why the Watch might be triggered, so probably best if i update this to sort the Deployments before looping over them.

Reply:
I added the sort.